src/routes/OpportunityPage/OpportunityPage.js

```diff
@@ -149,6 +147,14 @@ class OpportunityPage extends Component {
 
   // this runs before the "render and return ( ... ) " runs. We use it to get data from the backend about the opportunity
   componentWillMount() {
+    axios.get(`/api/undergrads/star?type=opportunity&token_id=${sessionStorage.getItem('token_id')}`)
+    .then((response) => {
+      let data = response.data;
+      this.setState({ starred: data.includes(this.getId()) });
```

Comment:
I don't think you wrote `getId()`, but looking at that function I wonder if we just could have gotten the same result by doing `this.props.match.params.id` like on line 158 of this file, since in `index.js` (our router file) we define id as a wildcard operator (like in that get method in the backend). I tested the two functions/methods and they produce the same result.
```suggestion
      this.setState({ starred: data.includes(this.props.match.params.id) });
```